WalletWasabi.Documentation/WasabikaCodex.md

```diff
@@ -91,3 +92,131 @@ __Fact__: code review is one of the few scientifically proven code improvement t
   You can use it with the console in the project library by the following commands:  
    - `git upd` -> Update the forked repository  
    - `git del` -> Delete the current branch remotely and locally  
+
+# Productivity tips
+
+Developers work with many branches at the same time because during any single day they review others' pull requests, try new crazy things, work on new features/bug-fixes and also fix somebody else's PRs. All this forces a combination of `git stash [pop|save]`, `git fetch`, `git checkout [-f]` and again `git stash [pop|apply]`. No matter if this is done by command line or with a tool, the problem of having only one copy of the code is an ugly constrain.
+
+## Wasabi Wallet folders organization (suggestion)
+
+### Create main local bare repository
+
+First clone your existing repository as a bare repo (a repository without files) and do it in a `.git` directory.
+
+```
+$ git clone --bare git@github.com:zkSNACKs/WalletWasabi.git .git
+```
+
+## Working with remote repositories
+
+There are two non-mutually exclusive alternatives here. One is to work with the other developers' repositories directy or work with the `origin` repository only.
+
+### Work with devs remote repositories (alternative 1)
+
+Having the team members' repositories as remote makes the work easier:
+
+```
+$ git remote add dan git@github.com:danwalmsley/WalletWasabi.git
+$ git remote add jmacato git@github.com:jmacato/WalletWasabi.git
+$ git remote add kiminuo git@github.com:kiminuo/WalletWasabi.git
+$ git remote add max git@github.com:MaxHillebrand/WalletWasabi.git
+$ git remote add molnard git@github.com:molnard/WalletWasabi.git
+$ git remote add nopara73 git@github.com:nopara73/WalletWasabi.git
+$ git remote add yahia git@github.com:yahiheb/WalletWasabi.git
+$ git remote add yuval git@github.com:nothingmuch/WalletWasabi.git
+```
+
+Once we have all the remote repositories we fetch them all (this can be a bit slow but it takes time only the first time)
+
+```
+$ git fetch --all
+```
+
+After that we have absolutely everything locally.
+
+### Work with origin remote repository only (alternative 2)
```

Comment:
@soosr's `git pr` and my `git upd` commands are terribly useful, could be good addendum.

Reply:
What are those commands?
We have this `git upd` command: https://github.com/yahiheb/WalletWasabi/blob/productivity-worktree/WalletWasabi.Documentation/WasabikaCodex.md#git-tips

Reply:
```
pr = "!f(){ git checkout -b $(date +'%s') && git add . && git commit -S -m \"$1\" && git push -u origin $(git branch --show) && start \"$(echo $(git config --get remote.upstream.url) | sed 's/.git$//g')/compare/master...nopara73:$(git branch --show)?expand=1\"; };f"
```

1. You work on a branch.
2. `git pr "fix codefactor"`
3. It'll create a new branch, commit your work and open your browser with a new PR waiting for you ready to fill the title and the description.

Reply:
`/compare/master...nopara73:$(git branch --show)` Each one should use their github username here?

Reply:
Yes.

Reply:
Add this to the docs and we are ready.

Reply:
Done https://github.com/zkSNACKs/WalletWasabi/pull/5068/commits/6fbbd86035f2018579bc09d2040d48c636ac3f49